File: ai_calendar_app/lib/models/event.dart
class CalendarEvent {
  final int? id;
  final String title;
  final String? description;
  final DateTime startTime;
  final DateTime endTime;
  final String? location;
  final String category;
  final String priority;
  final String status;
  final String source;

  CalendarEvent({
    this.id,
    required this.title,
    this.description,
    required this.startTime,
    required this.endTime,
    this.location,
    this.category = 'other',
    this.priority = 'medium',
    this.status = 'confirmed',
    this.source = 'manual',
  });

  factory CalendarEvent.fromJson(Map<String, dynamic> json) => CalendarEvent(
        id: json['id'],
        title: json['title'] ?? '',
        description: json['description'],
        startTime: DateTime.parse(json['start_time']),
        endTime: DateTime.parse(json['end_time']),
        location: json['location'],
        category: json['category'] ?? 'other',
        priority: json['priority'] ?? 'medium',
        status: json['status'] ?? 'confirmed',
        source: json['source'] ?? 'manual',
      );

  Map<String, dynamic> toJson() => {
        if (id != null) 'id': id,
        'title': title,
        if (description != null) 'description': description,
        'start_time': startTime.toIso8601String(),
        'end_time': endTime.toIso8601String(),
        if (location != null) 'location': location,
        'category': category,
        'priority': priority,
        'status': status,
        'source': source,
      };

  CalendarEvent copyWith({
    int? id,
    String? title,
    String? description,
    DateTime? startTime,
    DateTime? endTime,
    String? location,
    String? category,
    String? priority,
    String? status,
    String? source,
  }) =>
      CalendarEvent(
        id: id ?? this.id,
        title: title ?? this.title,
        description: description ?? this.description,
        startTime: startTime ?? this.startTime,
        endTime: endTime ?? this.endTime,
        location: location ?? this.location,
        category: category ?? this.category,
        priority: priority ?? this.priority,
        status: status ?? this.status,
        source: source ?? this.source,
      );
}

File: backend/schemas/event.py
from datetime import datetime

from pydantic import BaseModel, Field


class EventCreate(BaseModel):
    title: str = Field(..., max_length=255)
    description: str | None = None
    start_time: datetime
    end_time: datetime
    location: str | None = None
    category: str = "other"
    priority: str = "medium"
    status: str = "confirmed"
    source: str = "manual"


class EventUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    start_time: datetime | None = None
    end_time: datetime | None = None
    location: str | None = None
    category: str | None = None
    priority: str | None = None
    status: str | None = None


class EventOut(BaseModel):
    id: int
    user_id: int
    title: str
    description: str | None = None
    start_time: datetime
    end_time: datetime
    location: str | None = None
    category: str
    priority: str
    status: str
    source: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class EventUpsertWS(BaseModel):
    temp_id: str
    title: str
    start_time: datetime
    end_time: datetime
    location: str | None = None
    category: str = "other"
    priority: str = "medium"
    status: str = "tentative"
    description: str | None = None


class EventConfirmWS(BaseModel):
    temp_id: str
    confirmed: bool
